Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/util/factory.go

```diff
@@ -108,11 +110,15 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
 	flags.SetNormalizeFunc(util.WarnWordSepNormalizeFunc) // Warn for "_" flags
 
-	generators := map[string]kubectl.Generator{
-		"run/v1":     kubectl.BasicReplicationController{},
-		"run-pod/v1": kubectl.BasicPod{},
-		"service/v1": kubectl.ServiceGeneratorV1{},
-		"service/v2": kubectl.ServiceGeneratorV2{},
+	generators := map[string]map[string]kubectl.Generator{
```

Comment:
I think a solution is to isolate the buckets and return them to the corresponding commands using separate calls. Generators vary a lot between them and having separate calls for separate groups makes sense to me.